Initial improvements to upgrade guide (#4119)
canonical-web-and-design/vanilla-framework: templates/docs/upgrade-guide-v3.md

The default value of `$breakpoint-navigation-threshold` was previously set to `$breakpoint-medium` and is now `$breakpoint-small`. This value should be overridden in project code to adjust the threshold at which navigation switches to dropdown, based on the number of navigation items.

In places where the `u-hide--small` class is currently used, `u-hide--medium` may need to be added. This is to replicate current behaviour of `u-hide--small`, ensuring the element is hidden on small and medium screens.

## Buttons

### Neutral buttons have been removed

The neutral button style provided by `p-button--neutral` was exactly the same as the default `p-button` styling, so it has been removed. Please use `p-button` or class-less `<button>` element instead.

### Full width mobile buttons have been removed

> In Vanilla 2.0 buttons on smallest screens (below `$breakpoint-x-small`) have been extended to full 100% width. This was causing issues on some layouts and with some utilities, so is no longer part of Vanilla 3.0. If your project has any custom code that was needed because of full width buttons you may need to update or remove it.

Reviewer: ```suggestion
In Vanilla 2.0 buttons on smallest screens (below `$breakpoint-x-small`) were extended to full 100% width. This was causing issues on some layouts and with some utilities, so is no longer part of Vanilla 3.0. If your project has any custom code that was needed because of full width buttons you may need to update or remove it.
```

## Inline images

We removed the inline images (`p-inline-images`) component. Please use the [logo section component](/docs/patterns/logo-section) instead. As this pattern requires images to have identical dimensions, the height and width of any which currently use the `p-inline-mages` pattern may need to be updated.

## Navigation

### Navigation class names have been updated

The `.row` class should not be used inside global navigation pattern. Use `.p-navigation__row` for fixed-width or `p-navigation_row--full-width` for full-width navigation.

Navigation classes `.p-navigation__links`, `.p-navigation__link`, and classless `<a>` have been removed. Please use new class names `.p-navigation__items`, `.p-navigation__item`, `.p-navigation__link` instead.

| Removed classes           | Replaced by            |
| ------------------------- | ---------------------- |
| `.p-navigation__links`    | `.p-navigation__items` |
| `.p-navigation__link`     | `.p-navigation__item`  |
| `.p-navigation__link > a` | `.p-navigation__link`  |

<div class="p-notification--caution">
  <div class="p-notification__content">

>     <p class="p-notification__message">Please note that <code>.p-navigation__link</code> class name is used in both deprecated and new version, but on different elements. When upgrading make sure not to replace valid new uses of <code>.p-navigation__link</code> applied to links (<code>&lt;a&gt;</code>), but only old ones applied to list items (<code>&lt;li&gt;</code>).</p>

Reviewer: ```suggestion
    <p class="p-notification__message">Please note that <code>.p-navigation__link</code> class name is used in both deprecated and new versions, but on different elements. When upgrading make sure not to replace valid new uses of <code>.p-navigation__link</code> applied to links (<code>&lt;a&gt;</code>), but only old ones applied to list items (<code>&lt;li&gt;</code>).</p>
```
